### ISSUE TITLE
Failing to raise undeclared error on set only accessor

### ISSUE THREAD

**Issue by csaltos** [microsoft/TypeScript#43203]
# Bug Report

If a property has a set only accessor then TypeScript fails on report an undeclared get accessor.

### 🔎 Search Terms

accessor undeclared error filtered by bug only

### 🕗 Version & Regression Information

TypeScript 4.2.3 ... but also fails for TypeScript 3.5.1 and the Nightly version as 4.3.0-dev.20210311

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.2.3#code/MYGwhgzhAEDCD2BXAdgFwKYCdoG8BQ0hA+qvKmCNALzQAMBhE6q0wSaAFAG4Bc0yiALYAjLAEpcDQtNQALAJYQAdCTIVoAahpcA3FIC+eQ3jbIILNigzYaydAHc47axzF7TEeCHRKQ8AOYclmhYSsGobtAA9FHQcorQELJIIAAm0KLQYMjQWJjwmHhAA)

### 💻 Code

```ts
class Counter {
   _total = 0
   set count(v: number) {
       this._total += v;
   }
}

const counter = new Counter();
console.log(counter.count); // this should be an error
```

### 🙁 Actual behavior

The code compiles but as it's obvious it reports an undefined valued (sadly TypeScript is not able to catch this error on compile time as its goal is in the first place)

### 🙂 Expected behavior

The code does not compile until the error of using the undeclared accesor is fixed.

**Comment by MartinJohns** [comment]
This issue exists for a long time already. It was previously closed and locked: #295

**Comment by csaltos** [comment]
So sad to hear that TypeScript cannot prevent this kind of clumsy errors ... but anyway, I hope one day it will.

Thanks for the quick reply !! :+1: :sunglasses:
